projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.spec.ts

```diff
@@ -65,25 +66,36 @@ describe('FhirChartSummaryComponent', () => {
 
   it('should render a card for each layer', () => {
     const layer: ManagedDataLayer = { id: '1', name: 'layer', datasets: [], scale: { id: 'test' } };
-    layerManager.selectedLayers$.next([layer, layer, layer]);
+    layerManager.enabledLayers$.next([layer, layer, layer]);
     fixture.detectChanges();
     const cards = fixture.debugElement.queryAll(By.directive(MockFhirChartSummaryCardComponent));
     expect(cards.length).toBe(3);
   });
 
   it('should set inputs on fhir-chart-summary-card component', () => {
     const layer: ManagedDataLayer = { id: '1', name: 'layer', datasets: [{ label: 'dataset', data: [] }], scale: { id: 'test' } };
-    layerManager.selectedLayers$.next([layer]);
+    layerManager.enabledLayers$.next([layer]);
     fixture.detectChanges();
     const card: DebugElement = fixture.debugElement.query(By.directive(MockFhirChartSummaryCardComponent));
     expect(card.componentInstance.title).toEqual('layer');
   });
 
   it('should set inputs on dynamic-table component', () => {
     const layer: ManagedDataLayer = { id: '1', name: 'layer', datasets: [{ label: 'dataset', data: [] }], scale: { id: 'test' } };
-    layerManager.selectedLayers$.next([layer]);
+    layerManager.enabledLayers$.next([layer]);
     fixture.detectChanges();
     const statistics: DebugElement = fixture.debugElement.query(By.directive(MockDynamicTableComponent));
     expect(statistics.componentInstance.data).toEqual([{ name: 'summary' }]);
   });
+  it('should not render a card for disabled layers', () => {
+    const enabledLayer: ManagedDataLayer = { id: '1', name: 'layer', datasets: [], scale: { id: 'test' },enabled:true };
+    const disabledLayer: ManagedDataLayer = { id: '1', name: 'layer', datasets: [], scale: { id: 'test' },enabled:false };
+    layerManager.selectedLayers$.next([enabledLayer, disabledLayer]);
+    layerManager.selectedLayers$.pipe(map((layers) => layers.filter((layer) => layer.enabled))).forEach(layer => {
+      layerManager.enabledLayers$.next(layer)
+    });
```

Comment:
This is really confusing. The test should not be subscribing to a mock observable. This component no longer depends on selectedLayers$ so I don't understand why the test is mocking that.

It also should not be duplicating the enabledLayers$ logic from the real code. It seems like this test is really only testing itself.

Reply:
Since you moved the logic for checking enabled layers into LayerManagerService, you should probably add an enabledLayers$ test in that class and delete this one.